ARCHITECTURE.md

```diff
@@ -60,12 +52,43 @@ Rojo has many data structures and their purpose might not be immediately clear a
 
 To learn more, read about [`memofs` architecture](crates/memofs/ARCHITECTURE.md).
 
+### LiveServer
+
+LiveServer underlies the [`serve` command](#the-serve-command) and provides the web server which clients (such as the plugin) can use to interface with [`ServeSession`](#servesession).
+
+There's two parts to the API, the UI and the API clients use.
+
+The UI provides information about the current [`DOM`](#rojotree) has, including metadata. It also shows the project name, up-time, and version its Rojo is on.
+
+The API provides a simple JSON protocol to interact with and receive changes from the underlying [`ServeSession`](#servesession). Checkout the [`api.rs` file under the web module](src/web/api.rs) to learn more.
+
 ### ServeSession
 
+The linchpin of Rojo. It contains all of the required components to serve a given project file.
```

Comment:
```suggestion
The `ServeSession` is the core of Rojo. It contains all of the required components to serve a given project file.
```

As much as I like the word linchpin, I feel like we should prioritize clarity here.